irmaclient/session.go

```diff
@@ -170,20 +182,20 @@ func (client *Client) newManualSession(request irma.SessionRequest, handler Hand
 	return session
 }
 
-// newQrSession creates and starts a new interactive IRMA session
-func (client *Client) newQrSession(qr *irma.Qr, handler Handler) SessionDismisser {
+// NewQrSession creates and starts a new interactive IRMA session
+func (client *Client) NewQrSession(qr *irma.Qr, handler Handler) (irma.ClientAuthorization, SessionDismisser) {
```

Comment:
In some way I need the ClientAuthorization to be available in the tests. There are some tests that test whether endpoints can be re-fetched and succeed or fail depending on when that should be possible or not. If I don't send it we know for sure the request is denied due to sending an unauthorized request. I also thought it was a bit strange to include this in the handler, because then also the irma bridge has to implement it and it does not even need this authorization code. 

The `Qr` reference of course comes from that this function receives the Qr as an object instead of as string. If you know something better to solve this problem than I'm happy to change it.

Reply:
I can think of the following potential solutions:

* Move the session test machinery to the `internal/test` package, so you can invoke them from within the `irmaclient` unittests. If `dismisser := NewSession()` then there you can do `dismisser.(*session).transport.headers` and fetch the token. However, this would be a giant refactor of the testing code, and this will not solve the other similar problem (where you use `SetMaxVersion` to execute existing unittests with a client pretending to not support session binding).
* Keep the function public but indicate in its name that it's for testing by prepending e.g. `Test`.
* Make the function accept a struct defined somewhere within `internal` as a (dummy) parameter. That would make it impossible to invoke it from anyplace outside of `irmago`.
* Access the private fields `dismisser.transport.headers` using reflection from within the `sessiontest` package, see https://stackoverflow.com/a/43918797. In this fashion it is also possible to write to private fields, so this can also solve the other case, if you make `maxVersion` a member of the `client` struct.

I think I like the last option best: It's not pretty but it solves the problem completely, and in terms of the amount of work it is very feasible. Additionaly, generally it's not strange for a unit test to access private variables - it is made strange here only because the test cannot live within the `irmaclient` package itself.